Task 4 completed

awestor opened the pull request:
All the main entities have been added for further implementation of the project. 6 repositories using JPA have been developed, as well as 1 interface and 1 repository class implementing it using Criteria API. Services have been implemented: 1 main service, and the rest are temporarily auxiliary. The main service demonstrates a method with manual transaction control. Autotests have been written for the implemented methods of the Product entity repository, as well as its variant using Criteria API. Tests were also written for a service that implements manual control over transactions.

awestor committed:
Task 4 completed

All the main entities have been added for further implementation of the project. 6 repositories using JPA have been developed, as well as 1 interface and 1 repository class implementing it using Criteria API. Services have been implemented: 1 main service, and the rest are temporarily auxiliary. The main service demonstrates a method with manual transaction control. Autotests have been written for the implemented methods of the Product entity repository, as well as its variant using Criteria API. Tests were also written for a service that implements manual control over transactions.

LARCH1K reviewed (commented):
В целом всё отлично, очень много сделано. Есть мелкие замечания - исправь их перед мерджем. На повторную проверку отправлять не нужно.

Также давай сразу избавимся от консольного ввода и всего, что к нему относится - в дальнейшем он нам не понадобится

awestor committed:
Correction of comments

Test methods were moved from the deleted class to ProductRepositoryTest. Classes and their files were removed to implement console input. Data access to the database was implemented using launch arguments.
